daemon/cluster/cluster.go

```diff
@@ -1108,7 +1108,7 @@ func (c *Cluster) CreateService(s types.ServiceSpec, encodedAuth string) (*apity
 		if err != nil {
 			logrus.Warnf("unable to pin image %s to digest: %s", ctnr.Image, err.Error())
 			resp.Warnings = append(resp.Warnings, fmt.Sprintf("unable to pin image %s to digest: %s", ctnr.Image, err.Error()))
-		} else {
+		} else if ctnr.Image != digestImage {
```

Comment:
I noticed this message while testing and found it quite strange and misleading. It's of the form:

```
pinning image busybox:latest@sha256:29f5d56d12684887bdfa50dcd29fc31eea4aaf4ad3bec43daf19026a7ce69912 by digest: busybox:latest@sha256:29f5d56d12684887bdfa50dcd29fc31eea4aaf4ad3bec43daf19026a7ce69912
```

I think it would be okay to have a debug log message noting that pinning by digest is being skipped because the image reference already has a digest. But saying that we're doing digest pinning when we're actually not touching the image reference is misleading.

"service update" already skips this log message when there is no change to the reference. I thought it was an oversight that "service create" logs it anyway.